File: Lab 2/Ireni_100657302/kafka-python/consumer.py
from kafka import KafkaConsumer
import json, sys


def run():

    try:

        consumer = KafkaConsumer(
            "Users",             # topic name, brokers
            auto_offset_reset='earliest',
            bootstrap_servers=['localhost:9093',
                               'localhost:9094', 'localhost:9095']
        )

        consumer.subscribe(['Users'])

#         infinite loop
        for msg in consumer:
            print(msg)

    except Exception as e:
        print("Something bad happened " + str(e))

    except KeyboardInterrupt as e:
        sys.exit(0)

run()

File: Lab 2/Ireni_100657302/kafka/consumer.js
//acquire library
const Kafka = require("kafkajs").Kafka

run();

async function run() {

    try {
        //create an admin connection to create a topic
        //establish tcp connection

        const kafka = new Kafka({
            "clientId": "myapp",
            "brokers": ["localhost:9093"]//,"localhost:9094","localhost:9095"]  
        })

        //admin interface
        const consumer = kafka.consumer({ "groupId": "test" });
        console.log("Connecting...")
        await consumer.connect()
        console.log("Connected!")

        //does long polling
        // await consumer.disconnect();

        consumer.subscribe({
            "topic": "Users",
            "fromBeginning": true
        })

        await consumer.run({
            "eachMessage": async result => {
                console.log(`Received msg ${result.message.value} on parition ${result.partition}`)
            }
        })

    } catch (excecption) {
        console.error(`Something bad happened ${ex}`);
    }

    finally {
    }
}
